ambry-server/src/main/java/com/github/ambry/server/AmbryServerRequests.java

```diff
@@ -439,14 +450,39 @@ private AdminResponse handleHealthCheckRequest(DataInputStream requestStream, Ad
         if (hostBlobStore == null || !hostBlobStore.isStarted() ||
             ((hostBlobStore.getCurrentState() != ReplicaState.STANDBY) &&
                 (hostBlobStore.getCurrentState() != ReplicaState.LEADER))) {
-          hostHealthy = false;
+          serverHealthStatus = ServerHealthStatus.BAD;
           break;
```

Comment:
instead of breaking on first bad partition, how about iterate over all partitions and get an entire list of all bad partitions ?

Reply:
You have a point. It's hard to distinguish if the issue is from the replica bootstrapping or from the disk's hardware if we don't have both pieces of information. Let me ask the media-sre team just to confirm

Reply:
I changed the logic to store the non-standby/leader partitions as well. Thanks for that